src/panels/config/automation/trigger/types/ha-automation-trigger-device.ts

```diff
@@ -159,12 +164,27 @@ export class HaDeviceTrigger extends LitElement {
     });
   }
 
-  private _extraFieldsComputeLabelCallback(localize) {
+  private _extraFieldsComputeLabelCallback(localize, trigger: DeviceTrigger) {
     // Returns a callback for ha-form to calculate labels per schema object
-    return (schema) =>
+    return (schema): string =>
       localize(
```

Comment:
Shouldnt the order be reversed? The component should be able to override the default translations I think?

Reply:
Yes, that makes sense. Do you know where these default translations are used? 
```json
"extra_fields": {
                    "above": "Above",
                    "below": "Below",
                    "for": "Duration (optional)",
                    "zone": "[%key:ui::panel::config::automation::editor::triggers::type::zone::label%]"
                  },
```
I would think of `sensor` domain, but for the "zone" key I'm not sure. Should it maybe be moved to backend (like `components/sensor/strings.json`)?

Reply:
Yes, it should be moved to the backend.

Reply:
I moved all of this to the backend and removed the translations from the frontend. https://github.com/home-assistant/core/pull/115892/commits/c20bd3264fd5c96be617fbeb6fa4f1f556e86106

Also moved the `ComputeLabelCallback` functions to a common module so it can be used by conditions and actions too.